polkadot/consensus/src/offline_tracker.rs

```diff
@@ -38,31 +42,32 @@ impl Observed {
 		}
 	}
 
-	fn note_round_end(&mut self, was_online: bool) {
-		let now = Instant::now();
-
+	fn note_round_end(&mut self, now: Instant, was_online: Option<bool>) {
```

Comment:
we provide `now` in order to ensure that all rounds in the same block are considered contemporaneous, which lets us group which validators were offline for the previous block specifically.